### PR TITLE
Adds Flask and Fast APIs

### PR DESCRIPTION
**Pull request by rachfop**
## What was changed
Adds Flask and FastAPI samples to the samples repo.
## Why?

Shows how to create durable workflows with Flask and FastAPI.

## Checklist

1. Closes 

2. How was this tested:

```
poe [test, lint, lint-types, format]
black .
isort .
```

3. Any docs updates needed?

Added readme's in each directory.
Updated main readme with reference.

### CODE REVIEW

File: tests/fastapi/run_fast_test.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import uuid
```
Comment (cretz) [inline tests/fastapi/run_fast_test.py:1]: If you're gonna put the example at `fast_api`, then this dir should be `tests/fast_api`

File: tests/flask/test_flask_route.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import pytest
```
Comment (cretz) [inline tests/flask/test_flask_route.py:1]: If you're gonna put the example at `flask_api`, then this dir should be `tests/flask_api`

File: flask_api/run_flask.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import asyncio
+
+from flask import Flask
+from temporalio.client import Client
+
+# Import the workflow from the previous code
```
Comment (Sushisource) [inline flask_api/run_flask.py:6]:
```suggestion
```

I'd just get rid of this. Sort of a non-comment.

File: fast_api/readme.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+. venv/bin/activate
+```
+
+Install the dependencies:
+
+```bash
+pip install -r requirements.txt
+```
+
+Start the Temporal server:
```
[the remark this answers is not in the document]
Comment (cretz) [reply fast_api/readme.md]: If you're on Windows: https://stackoverflow.com/questions/1793735/change-case-of-a-file-on-windows

File: README.md [hunk 5]
```diff
@@ -10,12 +10,20 @@ The Python SDK is under development. There are no compatibility guarantees nor p
 
 Prerequisites:
 
-* Python >= 3.7
+* [Python](https://www.python.org/downloads/) >= 3.7
 * [Poetry](https://python-poetry.org)
-* [Local Temporal server running](https://docs.temporal.io/clusters/quick-install/)
+* [Local Temporal server running](https://github.com/temporalio/temporal)
```
Comment (cretz) [inline README.md:15]: Why change the link? Shouldn't we ask the do the quick install approach? And if that's not good enough for local quick install, can we change those docs?

File: README.md [hunk 6]
```diff
@@ -10,12 +10,20 @@ The Python SDK is under development. There are no compatibility guarantees nor p
...
+## Get started
 
+Start the Temporal server:
+
+    temporal server start-dev
+
+With this repository cloned: create and activate a virtual environment, and install dependencies at the root of the directory:
+
+    python3 -m venv venv
+    . venv/bin/activate
```
Comment (cretz) [inline README.md:26]: Poetry does the `venv` stuff for you

File: README.md [hunk 7]
```diff
@@ -10,12 +10,20 @@ The Python SDK is under development. There are no compatibility guarantees nor p
...
 * [Poetry](https://python-poetry.org)
-* [Local Temporal server running](https://docs.temporal.io/clusters/quick-install/)
+* [Local Temporal server running](https://github.com/temporalio/temporal)
 
-With this repository cloned, run the following at the root of the directory:
+## Get started
 
+Start the Temporal server:
+
+    temporal server start-dev
```
Comment (cretz) [inline README.md:21]: Our docs should probably be best place to tell how to start local server

File: fast_api/readme.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
+# FastAPI Temporal Workflow Template
+
+This repository provides settings for [FastAPI](https://fastapi.tiangolo.com) and the Temporal Python SDK for durable workflows.
+
+## Getting Started
+
+To use this template, follow these steps.
```
Comment (cretz) [inline fast_api/readme.md:7]:
```suggestion
For this sample, the optional `fast_api` dependency group must be included. To include, run:

    poetry install --with sentry

To run the sample, first see [README.md](https://github.com/temporalio/samples-python/blob/main/README.md) for prerequisites. Then follow the below steps.
```

File: fast_api/readme.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+Start the Worker:
+
+```bash
+python3 run_worker.py
```
Comment (cretz) [inline fast_api/readme.md:12]: I think they should use `poetry run python` which ensures they are in the virtual env with the right dependencies.

File: pyproject.toml [hunk 10]
```diff
@@ -43,6 +43,14 @@ opentelemetry-exporter-jaeger-thrift = "^1.13.0"
 optional = true
 dependencies = { sentry-sdk = "^1.11.0" }
 
+[tool.poetry.group.flask_api]
```
Comment (cretz) [inline pyproject.toml:46]: Pedantic, but please keep alphabetical order (please move this below `fast_api` and both above `open_telemetry`)

Comment (cretz) [inline pyproject.toml:46]: You'll need to update `.github/workflows/ci.yml` to add `--with fast_api,flask_api` as args to `poetry install`

File: tests/fast_api/run_fast_test.py [hunk 11]
```diff
@@ -0,0 +1,45 @@
+import uuid
```
Comment (cretz) [inline tests/fast_api/run_fast_test.py:1]: Just noticed this file name. We usually want all test files to start with `test_`.

**Review by cretz — APPROVED** [review]
LGTM. Minor nit on a file name, would like to see fixed but not required. Will watch CI.

File: tests/fast_api/test_fast_route.py [hunk 12]
```diff
@@ -0,0 +1,11 @@
+from fastapi.testclient import TestClient
+
+from fast_api.run_fast import app
```
Comment (cretz) [inline tests/fast_api/test_fast_route.py:3]: This is hardcoded to use `localhost:7233` which is not present in test cases. You'll need to change the hardcode or do something different

File: tests/fast_api/test_fast_route.py [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+    def get_address(self) -> str:
+        return "mock_server:1234"
+
+
+@pytest.fixture
+def server_address():
+    return MockServerAddress()
+
+
+def test_read_main(server_address):
```
Comment (cretz) [inline tests/fast_api/test_fast_route.py:30]: How is `server_address` expected to be used?

File: tests/fast_api/test_fast_route.py [hunk 14]
```diff
@@ -0,0 +1,15 @@
+import asyncio
+
+import aiohttp
+
+
+def test_read_main(client):
+    async def test_async():
```
Comment (cretz) [inline tests/fast_api/test_fast_route.py:7]: I think the outer function needs to be async and you don't need this inner function at all, just use the async stuff directly in the function

File: tests/fast_api/test_fast_route.py [hunk 15]
```diff
@@ -0,0 +1,15 @@
...
+        async with aiohttp.ClientSession() as session:
+            async with session.get(
+                f"http://{client.service_client.config.target_host}/"
```
Comment (cretz) [inline tests/fast_api/test_fast_route.py:10]: I wonder if you want to run on another port that's not port 80 for your sample and for this test

**Review by cretz — COMMENTED** [review]
You'll have to resolve the conflict before CI will run again

**Comment by cretz** [comment]
You have to format your code. Run `poe format` before committing.

**Comment by cretz** [comment]
@rachfop - Why closed/deleted? Samples didn't seem worth it or was there another reason?

**Pull request closed without merging**